device/src/radio/mod.rs

```diff
@@ -44,35 +43,53 @@ where
 
 use core::fmt;
 
-pub trait PhyRxTxBuf {
-    fn clear(&mut self);
-    fn extend(&mut self, buf: &[u8]);
-}
-
-impl<N> PhyRxTxBuf for Vec<u8, N>
-where
-    N: ArrayLength<u8>,
-{
-    fn clear(&mut self) {
-        self.clear();
-    }
-
-    fn extend(&mut self, buf: &[u8]) {
-        self.extend_from_slice(buf).unwrap();
-    }
-}
-
 pub trait PhyRxTx {
-    type PhyBuf: AsRef<[u8]> + AsMut<[u8]> + Default + PhyRxTxBuf;
     type PhyEvent: fmt::Debug;
     type PhyError: fmt::Debug;
     type PhyResponse: fmt::Debug;
 
     fn get_mut_radio(&mut self) -> &mut Self;
 
     // we require mutability so we may decrypt in place
-    fn get_received_packet(&mut self) -> &mut Self::PhyBuf;
+    fn get_received_packet(&mut self) -> &mut [u8];
     fn handle_event(&mut self, event: Event<Self>) -> Result<Response<Self>, Error<Self>>
     where
         Self: core::marker::Sized;
 }
+
+pub struct RadioBuffer<'a> {
+    packet: &'a mut [u8],
+    len: usize,
+}
+
+impl<'a> RadioBuffer<'a> {
+    pub fn new(packet: &'a mut [u8]) -> Self {
+        Self { packet, len: 0 }
```

Comment:
Isn't `len` actually `pos`?

Reply:
Yeah, i squinted a bit when reading that just now as well. Fixed